src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSession.cs

```diff
@@ -40,73 +34,33 @@ public class ProfilerSession
         public IXEventSession XEventSession { get; set; }
 
         /// <summary>
-        /// Try to set the session into polling mode if criteria is meet
-        /// </summary>
-        /// <returns>True if session set to polling mode, False otherwise</returns>
-        public bool TryEnterPolling()
-        {
-            lock (this.pollingLock)
-            {
-                if (pollImmediatly || (!this.isPolling && DateTime.Now.Subtract(this.lastPollTime) >= pollingDelay))
-                {
-                    this.isPolling = true;
-                    this.lastPollTime = DateTime.Now;
-                    this.pollImmediatly = false;
-                    return true;
-                }
-                else
-                {
-                    return false;
-                }
-            }
-        }
-
-        /// <summary>
-        /// Is the session currently being polled
+        /// Could events have been lost in the last poll
         /// </summary>
-        public bool IsPolling
+        public bool EventsLost
         {
             get
             {
-                return this.isPolling;
-            }
-            set
-            {
-                lock (this.pollingLock)
-                {
-                    this.isPolling  = value;
-                }
+                return this.eventsLost;
             }
         }
 
         /// <summary>
-        /// The delay between session polls
+        /// Functions for marking and retrieving the status of the ProfilerSession having an active XELite stream running. 
         /// </summary>
-        public TimeSpan PollingDelay
-        {
-            get
-            {
-                return pollingDelay;
-            }
+        public void toggleStreamLock(){
```

Comment:
Do you really want this to be a toggle instead of a setter? It seems like the scenarios this is being used for specifically want to turn streaming on/off and so toggling could result in unexpected state.